### PR TITLE
spec: update spec to include CRI changes

### PR DESCRIPTION
**Pull request by squeed**

### CODE REVIEW

File: spec/aci.md [hunk 1]
```diff
@@ -243,6 +243,8 @@ JSON Schema for the Image Manifest (app image manifest, ACI manifest), conformin
         * **port** (integer, required) port number that will be used; see also **count**. Must be >=1 and <=65535.
         * **count** (integer, optional, defaults to 1 if unset) specifies a range of ports, starting with "port" and ending with "port" + "count" - 1. Must be >=1.
         * **socketActivated** (boolean, optional, defaults to "false" if unsupplied) if set to true, the application expects to be [socket activated](http://www.freedesktop.org/software/systemd/man/sd_listen_fds.html) on these ports. To perform socket activation, the ACE MUST pass file descriptors using the [socket activation protocol](http://www.freedesktop.org/software/systemd/man/sd_listen_fds.html) that are listening on these ports when starting this app. If multiple apps in the same pod are using socket activation then the ACE must match the sockets to the correct apps using getsockopt() and getsockname().
+	* **userAnnotations** (string map, optional) Arbitrary metadata for end-user use. Must not affect runtime behavior.
```
Comment (jonboulle) [inline spec/aci.md]: Please update these to:

(object, optional) map of arbitrary key-value metadata for end-user use. Values must be strings.

Comment (jonboulle) [inline spec/aci.md]: Why "must not affect runtime behaviour"?

File: spec/pods.md [hunk 2]
```diff
@@ -186,3 +187,7 @@ JSON Schema for the Pod Manifest, conforming to [RFC4627](https://tools.ietf.org
 * **ports** (list of objects, optional) list of ports that SHOULD be exposed on the host.
     * **name** (string, required, restricted to the [AC Name](#ac-name-type) formatting) name of the port to be exposed on the host. This field is a key referencing by name ports specified in the Image Manifest(s) of the app(s) within this Pod Manifest; consequently, port names MUST be unique among apps within a pod.
     * **hostPort** (integer, required) port number on the host that will be mapped to the application port.
+    * **hostIP** (IP string, optional) IP address to forward from. If ommitted, all IPs will be used
+    * **podPort** (Object, optional) A Port object as specified in an Image Manifest. If supplied, this overrides any matching ports specified by the images.
```
Comment (jonboulle) [inline spec/pods.md]: (object, optional) 
port object of the same structure as in an image manifest [see the "ports" field in the [Image Manifest Schema](aci.md#image-manifest-schema))

File: spec/pods.md [hunk 3]
```diff
@@ -186,3 +187,7 @@ JSON Schema for the Pod Manifest, conforming to [RFC4627](https://tools.ietf.org
 * **ports** (list of objects, optional) list of ports that SHOULD be exposed on the host.
     * **name** (string, required, restricted to the [AC Name](#ac-name-type) formatting) name of the port to be exposed on the host. This field is a key referencing by name ports specified in the Image Manifest(s) of the app(s) within this Pod Manifest; consequently, port names MUST be unique among apps within a pod.
     * **hostPort** (integer, required) port number on the host that will be mapped to the application port.
+    * **hostIP** (IP string, optional) IP address to forward from. If ommitted, all IPs will be used
```
Comment (jonboulle) [inline spec/pods.md]: (string, optional) an IP address to forward from. If omitted, all IPs will be used. Must be of the form xxxxx

File: spec/pods.md [hunk 4]
```diff
@@ -186,3 +187,7 @@ JSON Schema for the Pod Manifest, conforming to [RFC4627](https://tools.ietf.org
 * **ports** (list of objects, optional) list of ports that SHOULD be exposed on the host.
     * **name** (string, required, restricted to the [AC Name](#ac-name-type) formatting) name of the port to be exposed on the host. This field is a key referencing by name ports specified in the Image Manifest(s) of the app(s) within this Pod Manifest; consequently, port names MUST be unique among apps within a pod.
```
Comment (jonboulle) [inline spec/pods.md]: You need to caveat this by referencing podPort now I think

**Comment by squeed** [comment]
@jonboulle thanks, updated.

File: spec/pods.md [hunk 5]
```diff
@@ -184,5 +185,9 @@ JSON Schema for the Pod Manifest, conforming to [RFC4627](https://tools.ietf.org
...
 * **ports** (list of objects, optional) list of ports that SHOULD be exposed on the host.
-    * **name** (string, required, restricted to the [AC Name](#ac-name-type) formatting) name of the port to be exposed on the host. This field is a key referencing by name ports specified in the Image Manifest(s) of the app(s) within this Pod Manifest; consequently, port names MUST be unique among apps within a pod.
+    * **name** (string, required, restricted to the [AC Name](#ac-name-type) formatting) name of the port to be exposed on the host. This field is a key referencing by name ports specified in the Image Manifest(s) of the app(s) within this Pod Manifest, unless overridden by podPort. Consequently, port names MUST be unique among apps within a pod.
     * **hostPort** (integer, required) port number on the host that will be mapped to the application port.
+    * **hostIP** (string, optional) an IPv4 address to forward from. If ommitted, all IPs will be used. Must be in dotted-quad notation, e.g. "203.0.113.23"
```
Comment (jonboulle) [inline spec/pods.md]: ommitted --> omitted

File: spec/pods.md [hunk 6]
```diff
@@ -167,8 +167,9 @@ JSON Schema for the Pod Manifest, conforming to [RFC4627](https://tools.ietf.org
     * **app** (object, optional) substitute for the app object of the referred image's ImageManifest. See [Image Manifest Schema](aci.md#image-manifest-schema) for what the app object contains.
     * **readOnlyRootFS** (boolean, optional, defaults to "false" if unsupplied) whether or not the root filesystem of the app will be mounted read-only.
     * **mounts** (list of objects, optional) list of mounts mapping an app mountPoint to a volume. Each mount has the following set of key-value pairs:
-      * **volume** (string, required) name of the volume that will fulfill this mount (restricted to the [AC Name](types.md#ac-name-type) formatting); this is a key into the list of `volumes`, below.
+      * **volume** (string, required) name of the volume that will fulfill this mount (restricted to the [AC Name](types.md#ac-name-type) formatting); this is a key into the list of `volumes`, below, unless overriden by appVolume.
```
Comment (jonboulle) [inline spec/pods.md]: overridden

File: spec/aci.md [hunk 7]
```diff
@@ -243,6 +243,8 @@ JSON Schema for the Image Manifest (app image manifest, ACI manifest), conformin
         * **port** (integer, required) port number that will be used; see also **count**. Must be >=1 and <=65535.
         * **count** (integer, optional, defaults to 1 if unset) specifies a range of ports, starting with "port" and ending with "port" + "count" - 1. Must be >=1.
         * **socketActivated** (boolean, optional, defaults to "false" if unsupplied) if set to true, the application expects to be [socket activated](http://www.freedesktop.org/software/systemd/man/sd_listen_fds.html) on these ports. To perform socket activation, the ACE MUST pass file descriptors using the [socket activation protocol](http://www.freedesktop.org/software/systemd/man/sd_listen_fds.html) that are listening on these ports when starting this app. If multiple apps in the same pod are using socket activation then the ACE must match the sockets to the correct apps using getsockopt() and getsockname().
+	* **userAnnotations** (object, optional) Map of arbitrary key-value data for end-user use. Values must be strings. Unlike Annotations, UserAnnotations MUST NOT affect runtime behavior and support arbitrary strings as keys.
```
Comment (jonboulle) [inline spec/aci.md]: Map -> map

File: spec/aci.md [hunk 8]
```diff
@@ -243,6 +243,8 @@ JSON Schema for the Image Manifest (app image manifest, ACI manifest), conformin
...
+	* **userLabels** (object, optional) Map of arbitrary key-value data for end-user use. Values must be strings.
```
Comment (jonboulle) [inline spec/aci.md]: Map -> map

File: spec/aci.md (hunk 7 above)
Comment (jonboulle) [inline spec/aci.md]: behavior, and

File: spec/pods.md [hunk 9]
```diff
@@ -167,8 +167,9 @@ JSON Schema for the Pod Manifest, conforming to [RFC4627](https://tools.ietf.org
...
       * **path** (string, required) path inside the app filesystem to mount the volume; generally this will come from one of an app's mountPoint paths. For example, if an app has a mountPoint named "work" with path "/var/lib/work", an executor should map an appropriate volume to fulfill that mountPoint by using a `mount` object with that path.
+      * **appVolume** (object, optional) Same type as the Volume object in the pod spec. If supplied, overrides finding the Volume object by name.
```
Comment (jonboulle) [inline spec/pods.md]: same

**Comment by squeed** [comment]
@jonboulle running out of red ink yet :-)?

The intended use case for labels is to act as searchable selectors. Is that appropriate to specify in the CRI?

**Comment by jonboulle** [comment]
I have some more minor copy edits but will fast follow myself to speed things up.

**Pull request merged**